examples/string-templates/string_templates.bal

```diff
@@ -0,0 +1,21 @@
+import ballerina/io;
+
+public function main() {
+    // Note first \n is treated as is.
+    string s1 = string `line one \n rest of line 1 ${"\n"} second line`;
+    io:println(s1);
+
+    // You can use interpolations to add ` and  $ characters.
+    string s2 = string `Backtick: ${"`"} Dollar: ${"$"}`;
+    io:println(s2);
+
+    // You can nest template expressions.
+    string s3 = string `outer ${string `inner ${5} inner rest`} rest`;
+    io:println(s3);
+
+    // You can use an empty string interpolation to break a template expression
+    // across multiple lines
```

Comment:
or break at interpolations.

Let's add examples with both.

Also need a fullstop at the end.

Reply:
I didn't fully understand this. Do you mean 
```
    string s4 = string `prefix ${
                ""}middle ${""  // <-
                }suffix`;
```

Reply:
I meant breaking at a normal interpolation like

```ballerina
string user = "John";
string s4 = string `Hello World, from ${
                user}!`
```